src/main/java/edu/harvard/iq/dataverse/engine/command/impl/AbstractCreateDatasetCommand.java

```diff
@@ -0,0 +1,168 @@
+package edu.harvard.iq.dataverse.engine.command.impl;
+
+import edu.harvard.iq.dataverse.*;
+import edu.harvard.iq.dataverse.authorization.Permission;
+import edu.harvard.iq.dataverse.authorization.users.AuthenticatedUser;
+import edu.harvard.iq.dataverse.dataaccess.DataAccess;
+import edu.harvard.iq.dataverse.datacapturemodule.DataCaptureModuleUtil;
+import edu.harvard.iq.dataverse.datacapturemodule.ScriptRequestResponse;
+import edu.harvard.iq.dataverse.engine.command.CommandContext;
+import edu.harvard.iq.dataverse.engine.command.DataverseRequest;
+import edu.harvard.iq.dataverse.engine.command.RequiredPermissions;
+import edu.harvard.iq.dataverse.engine.command.exception.CommandException;
+import edu.harvard.iq.dataverse.engine.command.exception.CommandExecutionException;
+import edu.harvard.iq.dataverse.settings.SettingsServiceBean;
+import static edu.harvard.iq.dataverse.util.StringUtil.isEmpty;
+import java.io.IOException;
+import java.util.Objects;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**;
+ * An abstract base class for commands that creates {@link Dataset}s.
+ *
+ * @author michael
+ */
+@RequiredPermissions(Permission.AddDataset)
+public abstract class AbstractCreateDatasetCommand extends AbstractDatasetCommand<Dataset> {
+    
+    private static final Logger logger = Logger.getLogger(AbstractCreateDatasetCommand.class.getCanonicalName());
+    
+    final protected boolean registrationRequired;
+    
+    public AbstractCreateDatasetCommand(Dataset theDataset, DataverseRequest aRequest) {
+        this(theDataset, aRequest, false);
+    }
+
+    public AbstractCreateDatasetCommand(Dataset theDataset, DataverseRequest aRequest, boolean isRegistrationRequired) {
+        super(aRequest, theDataset);
+        registrationRequired = isRegistrationRequired;
+    }
+   
+    protected void additionalParameterTests(CommandContext ctxt) throws CommandException {
```

Comment:
why not make this abstract?

Reply:
Doing nothing is the default, valid behavior. So either have a an empty method in some subclasses, or have a single one here.